### PR TITLE
Make colors configurable

### PR DESCRIPTION
**Pull request by scholtzan**
Line colors are now configurable and a few other adjustments. There are some hickups when the number of lines change but I'll open a separate bug for that. It can be resolved by reloading the page for now.

At some point we'll also need to do some cleanup/refactoring of the code :D

### CODE REVIEW

File: src/ci-line-chart.js [hunk 1]
```diff
@@ -207,6 +190,25 @@ const vis = {
...
+          label: label_text,
+          display: "color",
+          default: defaultColor
+        };
+      }
+    });
+
+    // register options with parent page to update visConfig
+    this.trigger('registerOptions', this.options);
+
```
Comment (scholtzan) [inline src/ci-line-chart.js:211]: Adds each line as an option a color can be assigned to.

File: src/ci-line-chart.js [hunk 2]
```diff
@@ -226,12 +228,17 @@ const vis = {
...
+              round: 'day',
             },
             title: {
               display: true,
               text: optionsToFriendly[config.field_x]
-            }
+            },
+            grid: {
+              display: false
+            },
```
Comment (scholtzan) [inline src/ci-line-chart.js:241]: I think we might want to make this configurable, but at least for opmon the default is to set this off

File: src/ci-line-chart.js [hunk 3]
```diff
@@ -226,12 +228,17 @@ const vis = {
           x: {
             type: 'time',
             time: {
-              tooltipFormat: 'DD T'
+              tooltipFormat: 'DD',
+              unit: 'day',
+              round: 'day',
```
Comment (scholtzan) [inline src/ci-line-chart.js:233]: I think we'd want to do some automated detection on whether the data are dates, timestamps etc. But for now to support opmon let's do date as default.

File: src/ci-line-chart.js [hunk 4]
```diff
@@ -256,16 +263,17 @@ const vis = {
...
+        label: 'Lower' + groupLabel,
         data: groupData,
         fill: 1,
         parsing: {
           yAxisKey: 'CI_left'
         },
-        borderColor: idx === 0 ? 'rgba(0,0,200,1)' : 'rgba(200,0,0,1)',
-        backgroundColor: idx === 0 ? 'rgba(0,0,200,0.5)' : 'rgba(200,0,0,0.5)',
+        backgroundColor: config[group] + "33",
+        borderColor: config[group],
```
Comment (scholtzan) [inline src/ci-line-chart.js:276]: use configured color and adjust opacity

File: src/ci-line-chart.js (hunk 3 above)
Comment (mikewilli) [reply src/ci-line-chart.js:233]: Yep, I believe this is covered by #11

File: src/ci-line-chart.js (hunk 4 above)
Comment (mikewilli) [reply src/ci-line-chart.js:276]: Great! We may want to add something to ensure the configured color is hex format. RGB is technically accepted by the color picker (though the user would have to type/paste it in manually).

File: src/ci-line-chart.js (hunk 2 above)
Comment (mikewilli) [reply src/ci-line-chart.js:241]: Added a ticket (#18)

File: src/ci-line-chart.js [hunk 5]
```diff
@@ -256,16 +263,17 @@ const vis = {
     Array.from(pivotFieldNames).forEach((group, idx) => {
       // Filter data to the current group
       const groupData = d3data.filter((d) => d.pivot === group);
+      const groupLabel = group != 'NONE' ? ` - ${group}` : '';
 
       cfg.data.datasets.push({
-        label: `Lower-${group}`,
+        label: 'Lower' + groupLabel,
```
Comment (mikewilli) [inline src/ci-line-chart.js:269]: Usually prefer templates over string concatenation. 
```
`Lower ${groupLabel}` // <-- prefer this
'Lower' + groupLabel  //     over this
```

File: src/ci-line-chart.js [hunk 6]
```diff
@@ -256,16 +263,17 @@ const vis = {
     Array.from(pivotFieldNames).forEach((group, idx) => {
       // Filter data to the current group
       const groupData = d3data.filter((d) => d.pivot === group);
+      const groupLabel = group != 'NONE' ? ` - ${group}` : '';
```
Comment (mikewilli) [inline src/ci-line-chart.js:266]:
```suggestion
      const groupLabel = group !== 'NONE' ? ` - ${group}` : '';
```
Prefer strict equality check over abstract equality.

File: src/ci-line-chart.js [hunk 7]
```diff
@@ -256,16 +263,17 @@ const vis = {
...
-        label: `Lower-${group}`,
+        label: 'Lower' + groupLabel,
         data: groupData,
         fill: 1,
         parsing: {
           yAxisKey: 'CI_left'
         },
-        borderColor: idx === 0 ? 'rgba(0,0,200,1)' : 'rgba(200,0,0,1)',
-        backgroundColor: idx === 0 ? 'rgba(0,0,200,0.5)' : 'rgba(200,0,0,0.5)',
+        backgroundColor: config[group] + "33",
```
Comment (mikewilli) [inline src/ci-line-chart.js:275]:
```suggestion
        backgroundColor: `${config[group]}33`,
```
Suggestion using template string

File: src/ci-line-chart.js [hunk 8]
```diff
@@ -277,14 +285,14 @@ const vis = {
         }
       });
       cfg.data.datasets.push({
-        label: `${optionsToFriendly[config.field_y]}-${group}`,
+        label: `${optionsToFriendly[config.field_y]}` + groupLabel,
```
Comment (mikewilli) [inline src/ci-line-chart.js:288]:
```suggestion
        label: `${optionsToFriendly[config.field_y]}${groupLabel}`,
```
Suggestion to include the groupLabel in the existing template

File: src/ci-line-chart.js [hunk 9]
```diff
@@ -296,14 +304,14 @@ const vis = {
         }
       });
       cfg.data.datasets.push({
-        label: `Upper-${group}`,
+        label: "Upper" + groupLabel,
```
Comment (mikewilli) [inline src/ci-line-chart.js:307]:
```suggestion
        label: `Upper${groupLabel}`,
```

File: src/ci-line-chart.js [hunk 10]
```diff
@@ -296,14 +304,14 @@ const vis = {
...
         data: groupData,
         fill: 1,
         parsing: {
           yAxisKey: 'CI_right'
         },
-        borderColor: idx === 0 ? 'rgba(0,0,200,1)' : 'rgba(200,0,0,1)',
-        backgroundColor: idx === 0 ? 'rgba(0,0,200,0.5)' : 'rgba(200,0,0,0.5)',
+        backgroundColor: config[group] + "33",
```
Comment (mikewilli) [inline src/ci-line-chart.js:313]:
```suggestion
        backgroundColor: `${config[group]}33`,
```

**Review by mikewilli — COMMENTED** [review]
Looks good to me -- made a couple small suggestions, but overall good to go!

**Pull request merged**